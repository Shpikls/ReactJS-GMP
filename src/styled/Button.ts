import styled from 'styled-components'

export const Button = styled.button`
	font-family: Montserrat, sans-serif;
	font-style: normal;
	border: none;
	border-radius: 5px;
	cursor: pointer;
	text-align: center;
	text-transform: uppercase;
`
