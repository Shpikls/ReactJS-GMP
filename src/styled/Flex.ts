import styled from 'styled-components'

type JustifyContent =
	| 'flex-start'
	| 'flex-end'
	| 'center'
	| 'space-between'
	| 'space-around'
	| 'space-evenly'
	| 'initial'
	| 'inherit'

export const Flex = styled.div<{ jContent?: JustifyContent }>`
	display: flex;
	${({ jContent }) => {
		if (jContent) return `justify-content: ${jContent}`
	}};
`
