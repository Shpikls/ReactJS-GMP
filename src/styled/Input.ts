import styled from 'styled-components'

export const InputText = styled.input.attrs({ type: 'text' })`
	width: 713px;
	height: 57px;
	box-sizing: border-box;
	opacity: 0.7;
	border-radius: 4px;
	background: rgba(50, 50, 50, 0.8);
	border: none;
	font-family: Montserrat, sans-serif;
	font-style: normal;
	font-weight: normal;
	font-size: 20px;
	line-height: 24px;
	color: #ffffff;
	align-items: center;
	padding-left: 20px;
`
