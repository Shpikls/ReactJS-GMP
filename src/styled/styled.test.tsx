import { render } from '@testing-library/react'
import React from 'react'
import { Button } from './Button'
import { Flex } from './Flex'
import { InputText } from './Input'

describe('Test styled components', () => {
	test('Button to mach snapshot', () => {
		const { asFragment } = render(<Button />)
		expect(asFragment()).toMatchSnapshot()
	})

	test('Flex to mach snapshot', () => {
		const { asFragment } = render(<Flex jContent="space-between" fDirection="column" />)
		expect(asFragment()).toMatchSnapshot()
	})

	test('Input to mach snapshot', () => {
		const { asFragment } = render(<InputText />)
		expect(asFragment()).toMatchSnapshot()
	})
})
