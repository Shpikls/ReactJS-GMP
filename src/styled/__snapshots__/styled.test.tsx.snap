// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test styled components Button to mach snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBQB btGkqL"
  />
</DocumentFragment>
`;

exports[`Test styled components Flex to mach snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTEea huIxuF"
  />
</DocumentFragment>
`;

exports[`Test styled components Input to mach snapshot 1`] = `
<DocumentFragment>
  <input
    class="sc-dlfnuX hrctIj"
    type="text"
  />
</DocumentFragment>
`;
